setup.py

```diff
@@ -256,10 +273,10 @@ def setup_package():
                     cmdclass=cmdclass,
                     python_requires=">=3.6",
                     install_requires=[
-                        'numpy>={}'.format(NUMPY_MIN_VERSION),
-                        'scipy>={}'.format(SCIPY_MIN_VERSION),
-                        'joblib>={}'.format(JOBLIB_MIN_VERSION),
-                        'threadpoolctl>={}'.format(THREADPOOLCTL_MIN_VERSION)
+                        f'numpy>={NUMPY_MIN_VERSION}',
+                        f'scipy>={SCIPY_MIN_VERSION}',
+                        f'joblib>={JOBLIB_MIN_VERSION}',
+                        f'threadpoolctl>={THREADPOOLCTL_MIN_VERSION}'
```

Comment:
When building from source, this would result in a `SyntaxError`:

```
    f'numpy>={NUMPY_MIN_VERSION}',
                                ^
SyntaxError: invalid syntax
```

rather than the nicer:

```
ERROR: Package 'scikit-learn' requires a different Python: 3.5.5
```

Reply:
I've reinstalled the source in a new virtual environment with the editable mode [as indicated in the documentation](https://scikit-learn.org/stable/developers/advanced_installation.html#building-from-source) using:
```bash
python -m venv /tmp/venv
source /tmp/venv/bin/activate
pip install wheel
pip install cython
pip install --verbose --no-build-isolation --editable .
```
but I haven't meet this problem.

How can I reproduce this problem?

Reply:
Thomas is right that f-stings are not supported in Python <=3.5: we should revert to using `.format` so that people who mistakenly use Py3.5 get a nice error message.

Reply:
As far as I've seen, scikit-learn requires python 3.6 and newer, isn't it?

Reply:
Yes, but if you mistakenly try install it for Python 3.5- previously one would get an error saying that you need python 3.6+ now it will just produce a syntax error which is not very nice.

Reply:
OK, make sense. I followed @glemaitre 's suggestions (see above) but I can revert back to the `format` syntax.